crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -339,21 +340,24 @@ unsafe impl<'w, 's, D: ReadOnlyQueryData + 'static, F: QueryFilter + 'static> Re
 // SAFETY: Relevant query ComponentId and ArchetypeComponentId access is applied to SystemMeta. If
 // this Query conflicts with any prior access, a panic will occur.
 unsafe impl<D: QueryData + 'static, F: QueryFilter + 'static> SystemParam for Query<'_, '_, D, F> {
-    type State = QueryState<D, F>;
-    type Item<'w, 's> = Query<'w, 's, D, F>;
+    type State = Entity;
+    type Item<'w, 's> = Query<'w, 'w, D, F>;
```

Comment:
Does the lifetime change here is correct? This is the main reason `bevy` crate is not compile. The ecs crate itself is already compiled and passing all the tests now. I think it I did it right here, so the temporary solution is doing the same to all places where the compiler is not happy. Then remove the lifetime in the follow up PR when we make the `QueryState` private.

Reply:
Yes this lifetime is correct.